src/components/Viewport.ts

```diff
@@ -1476,4 +1675,161 @@ export default class Viewport {
       } as Partial<NeedPanelEvent>,
     );
   }
+
+  private updateVisiblePanels() {
+    const state = this.state;
+    const options = this.options;
+    const isHorizontal = this.options.horizontal;
+    const cameraElement = this.cameraElement;
+    const visibleIndex = state.visibleIndex;
+
+    if (!options.renderOnlyVisible) {
+      return;
+    }
+
+    if (!this.nearestPanel) {
+      this.resetVisibleIndex();
+      while (cameraElement.firstChild) {
+        cameraElement.removeChild(cameraElement.firstChild);
+      }
+      return;
+    }
+
+    const panelManager = this.panelManager;
+    const allPanelCount = panelManager.getRange().max + 1;
+    const cloneCount = panelManager.getCloneCount();
+    const cameraPos = this.getCameraPosition();
+    const basePanel = this.nearestPanel;
+    const viewportBbox = this.getBbox();
+    const bbox = this.options.overflow
+      ? this.getWindowBBox()
+      : viewportBbox;
+    const viewportPos = isHorizontal
+      ? viewportBbox.x
+      : viewportBbox.y;
+
+    const checkLastPanel = (
+      panel: Panel,
+      getNextPanel: (panel: Panel) => Panel | null,
+      isOutOfViewport: (panel: Panel) => boolean,
+    ): Panel => {
+      let lastPanel = panel;
+      while (true) {
+        const nextPanel = getNextPanel(lastPanel);
+        if (!nextPanel || isOutOfViewport(nextPanel)) {
+          break;
+        }
+        lastPanel = nextPanel;
+      }
```

Comment:
How about following? (looks slightly simple?)
```ts

while (nextPanel = getNextPanel(lastPanel)) {
  if (isOutOfViewport(nextPanel)) {
    break;
  }
  lastPanel = nextPanel;
}
```

Reply:
That looks simple, but I heard that assigning value in while clause is not a good pattern.
Of course `while(true)` is not a good pattern either...